src/mlpack/core/util/version.hpp

```diff
@@ -20,6 +20,10 @@
 #define MLPACK_VERSION_MINOR 3
 #define MLPACK_VERSION_PATCH 2
 
+#define MLPACK_DATE_MONTH 04
+#define MLPACK_DATE_DAY 29
+#define MLPACK_DATE_YEAR 2020
```

Comment:
@rcurtin the addition of the release date is being used to set the package version date in `DESCRIPTION` via: 

https://github.com/Yashwants19/mlpack/pull/6/files#diff-c369a5bc9517d25df9cabd6008998030R51-R56

I think it could be grabbed from `git` and applied during the version release to generate version header defines.

Reply:
There is already some CMake hackery to set the version right when the code is in a git repository; maybe that could be expanded upon.  So, e.g., if you build the `mlpack_knn` command-line program and it's not a release, then the version is the git revision:

```
$ mlpack_knn --version
k-Nearest-Neighbors Search: part of mlpack git-398a27657.
```

Reply:
Nice. I used to do that from SVN too "way back when" but stopped because no human can make sense of sha1 hashes :)

Reply:
Agreed, but it makes bug reports a little bit easier :smile:

Reply:
Updated this. Used the same approach as used in `shogun toolbox` R interface.